packages/react-native-sdk/src/components/Livestream/LiveStreamControls/HostLiveStreamControls.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import {
+  HostStartStreamButton as DefaultHostStartStreamButton,
+  HostStartStreamButtonProps,
+} from './HostStartStreamButton';
+import {
+  LiveStreamMediaControls as DefaultLiveStreamMediaControls,
+  LiveStreamMediaControlsProps,
+} from './LiveStreamMediaControls';
+import { useTheme } from '../../../contexts';
+import { Z_INDEX } from '../../../constants';
+
+/**
+ * Props for the HostLiveStreamControls component.
+ */
+export type HostLiveStreamControlsProps = HostStartStreamButtonProps & {
+  /**
+   * Component to customize the host's start/end live stream button.
+   */
+  HostStartStreamButton?: React.ComponentType<HostStartStreamButtonProps> | null;
+  /**
+   * Component to customize the host's media control(audio/video) buttons.
+   */
+  LiveStreamMediaControls?: React.ComponentType<LiveStreamMediaControlsProps> | null;
+  /**
+   * Enable HTTP live streaming
+   */
+  hls?: boolean;
+};
+
+/**
+ * The HostLiveStreamControls component displays the call controls for the live stream at host's end.
+ */
+export const HostLiveStreamControls = ({
+  HostStartStreamButton = DefaultHostStartStreamButton,
+  LiveStreamMediaControls = DefaultLiveStreamMediaControls,
+  onEndStreamHandler,
+  onStartStreamHandler,
+  hls = false,
+}: HostLiveStreamControlsProps) => {
+  const {
+    theme: { colors, hostLiveStreamControls },
+  } = useTheme();
+  return (
+    <View
+      style={[
+        styles.container,
+        { backgroundColor: colors.static_overlay },
+        hostLiveStreamControls.container,
+      ]}
+    >
+      <View style={[styles.leftElement, hostLiveStreamControls.leftElement]}>
+        {HostStartStreamButton && (
+          <HostStartStreamButton
+            onEndStreamHandler={onEndStreamHandler}
+            onStartStreamHandler={onStartStreamHandler}
+            hls={hls}
+          />
+        )}
+      </View>
+      <View style={[styles.rightElement, hostLiveStreamControls.rightElement]}>
+        {LiveStreamMediaControls && <LiveStreamMediaControls />}
+      </View>
```

Comment:
```
        {LiveStreamMediaControls && <View style={[styles.rightElement, hostLiveStreamControls.rightElement]}><LiveStreamMediaControls /> </View>}
```

shouldnt it be like this? so that if null component is passed we don't see an empty spacing? 

can you check for everyother components also please

Reply:
So the thing here is according to designs, the media controls button should exist on the right end of the controls. Even  if user will tend to customize the view it should be added on right end corner. Hence it is done this way.